src/parser.js

```diff
@@ -1138,132 +1139,164 @@ export default class Parser {
     }
   }
 
-  // Convert a Swagger parameter into a Refract element.
-  convertParameterToElement(parameter, path, setAttributes = false) {
+  /* eslint-disable class-methods-use-this */
+  schemaForParameterValue(parameter) {
+    const schema = {
+      type: parameter.type,
+    };
+
+    if (schema.type === 'integer') {
+      schema.type = 'number';
+    }
+
+    if (parameter.items) {
+      schema.items = parameter.items;
+    }
+
+    return schema;
+  }
+
+  typeForParameter(parameter) {
     const {
       Array: ArrayElement, Boolean: BooleanElement, Number: NumberElement,
-      String: StringElement, Element,
+      String: StringElement,
     } = this.minim.elements;
 
-    let element;
-    let Type;
-    let example = parameter['x-example'];
-
-    // Convert from Swagger types to Minim elements
-    if (parameter.type === 'string') {
-      Type = StringElement;
-    } else if (parameter.type === 'integer' || parameter.type === 'number') {
-      Type = NumberElement;
-    } else if (parameter.type === 'boolean') {
-      Type = BooleanElement;
-    } else if (parameter.type === 'array') {
-      Type = ArrayElement;
-
-      if (example && !Array.isArray(example)) {
-        example = [];
-
-        this.createAnnotation(annotations.VALIDATION_WARNING, path.concat(['x-example']),
-          ('Value of example should be an array'));
-      }
-    } else {
-      // Default to a string in case we get a type we haven't seen
-      Type = StringElement;
-    }
+    const types = {
+      string: StringElement,
+      number: NumberElement,
+      integer: NumberElement,
+      boolean: BooleanElement,
+      array: ArrayElement,
+      file: StringElement,
+    };
+
+    return types[parameter.type];
+  }
 
-    const sampleContent = new Type(example);
+  createAnnotationsFromSchemaValidation(error) {
+    error.details.forEach((detail) => {
+      this.createAnnotation(annotations.VALIDATION_WARNING, this.path, detail.message);
+    });
+  }
+
+  // Convert a Swagger parameter into a Refract element.
+  convertParameterToElement(parameter, setAttributes = false) {
```

Comment:
This function is pretty long, it was before I started. I have made some improvements to the function layout, it is in three main sections and the isolation between them is further.

- Example
- Enum
- Default
- Array value from items if no example
- setAttribute

Each section really only knows about the parameter, validation of value and element now.